lib/galaxy/tools/xsd/galaxy.xsd

```diff
@@ -1990,6 +1990,33 @@ multiple formats.
        help="The set of aligned reads." />
 ```
 
+Perhaps counter-intuitively, a ``multiple="true"`` data parameter requires at least one
+data input. If ``optional="true"`` is specified, this condition is relaxed and the user
+is allowed to select 0 datasets. Unfortunately, if 0 datasets are selected the resulting
+value for the parameter during Cheetah templating (such as in a ``command`` block) will
+effectively a list with one ``None``-like entity in it.
+
+The following idiom can be used to iterate over such a list and build a hypothetical ``-B``
+parameter for each file - the ``if`` block is used to handle the case where a ``None``-like
+entity appears in the list because no files were selected: 
+
+```
+#for $input in $input1
+#if $input
+-B "$input"
+#end if
+#end for
+```
```

Comment:
```
#if str($input1) != 'None'
```

should be strongly be considered an anti-pattern in my opinion. It isn't the least bit intuitive or Pythonic and it is really tied down to the internals of Galaxy and these wrappers. It also assumes more from the templating language which makes the tool more tied to Cheetah and Galaxy. Templating elements should be as light-weight as possible to make the tool work.

The length thing doesn't work in hands:

```
cat: None: No such file or directory
```

Galaxy really is creating a list with a None-like thing in it - I think the length is 1 unfortunately.